lib/Serialization/Serialization.cpp

```diff
@@ -3726,10 +3726,28 @@ static void writeDeclCommentTable(
       if (!VD)
         return true;
 
-      // Skip the decl if it does not have a comment.
       RawComment Raw = VD->getRawComment();
-      if (Raw.Comments.empty() && !GroupContext.isEnable())
-        return true;
+
+      //when building the stdlib we intend to
+      //serialize unusual comments.  This situation is represented by
+      //GroupContext.isEnable().  In that case, we perform fewer serialization checks.
+      if (!GroupContext.isEnable()) {
+        // Skip the decl if it cannot have a comment.
+        if (!VD->canHaveComment()) {
+          return true;
+        }
+
+        // Skip the decl if it does not have a comment.
+
+        if (Raw.Comments.empty() && !GroupContext.isEnable())
+          return true;
+
+        // Skip the decl if it's not visible to clients.
+        if (!GroupContext.isEnable() && VD->getEffectiveAccess() != Accessibility::Public)
```

Comment:
Can you add a comment saying why `getEffectiveAccess` is correct here rather than `getFormalAccess`? That would normally be a mistake in Serialization.